llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp

```diff
@@ -1972,11 +1972,9 @@ static void fixFuncEntryCount(PGOUseFunc &Func, LoopInfo &LI,
                               BranchProbabilityInfo &NBPI) {
   Function &F = Func.getFunc();
   BlockFrequencyInfo NBFI(F, NBPI, LI);
-#ifndef NDEBUG
   auto BFIEntryCount = F.getEntryCount();
-  assert(BFIEntryCount && (BFIEntryCount->getCount() > 0) &&
-         "Invalid BFI Entrycount");
-#endif
+  if (!BFIEntryCount || BFIEntryCount->getCount() == 0)
```

Comment:
Currently the profile data does not track that it originated from sampled instrumentation code.  It could be added.  Alternatively to skipping fixFuncEntryCount (as implemented here) we could extend the many attempts to ensure the entry block count is never zero to cover the corner case shown in the added unit test.

Reply:
Yeah, given we're doing that already, I think it makes sense to fix it in `populateCounters` -- that code below should be close. Entry should not be zero if there is any count in the function. 

```
  // Fix the obviously inconsistent entry count.
  if (FuncMaxCount > 0 && FuncEntryCount == 0)
    FuncEntryCount = 1;
```